### PR TITLE
WIP: Fixes TealAppDriver & implement regression tests

### LINKED ISSUE #1535: Implement regression test
### Feature description

Related with https://github.com/insightsengineering/coredev-tasks/issues/589

### Scope of work

teal (40 hours)
- teal_data_module interaction
- teal_transform_module
- common data warnings and errors in teal app
- Check if the module can run, use example_module()
- Snapshot manager (check the modal shows with the right design)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [x] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [x] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by averissimo**
# Pull Request

Fixes #1535 

### Changes description

- [ ] Remove `testthat::skip("chromium")` from e2e tests
- [ ] Ensure all e2e pass
- [ ] Add regression tests

### CODE REVIEW

File: tests/testthat/test-shinytest2-data_summary.R [hunk 1]
```diff
@@ -85,62 +75,51 @@ testthat::test_that("e2e: data summary table is displayed with 3 columns for dat
...
 
-    app <- TealAppDriver$new(
-      data = data,
-      modules = example_module()
-    )
+  teal.data::join_keys(data) <- teal.data::join_keys(
+    teal.data::join_key("mtcars2", "mtcars1", keys = c("am"))
+  )
+
+  app <- TealAppDriver$new(data = data, modules = example_module())
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-data_summary.R:99]: Not sure what is happening, but I'm seeing a warning locally:
```
Warning (test-shinytest2-data_summary.R:99:3): e2e: data summary table does not list unsupported objects
'package:MultiAssayExperiment' may not be available when loading
```

File: tests/testthat/test-shinytest2-init.R [hunk 2]
```diff
@@ -2,18 +2,17 @@ testthat::skip_if_not_installed("shinytest2")
...
 
 testthat::test_that("e2e: teal app initializes with no errors", {
-  testthat::skip("chromium")
   skip_if_too_deep(5)
   app <- TealAppDriver$new(
     data = simple_teal_data(),
     modules = example_module(label = "Example Module")
   )
   app$expect_no_shiny_error()
+  app$expect_screenshot(selector = "#teal-tabpanel_wrapper")
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-init.R:11]: Locally I got a warning about missing windows snapshot (Not sure how important this is)

```
Warning (test-shinytest2-init.R:11:3): e2e: teal app initializes with no errors
Adding new file snapshot: 'tests/testthat/_snaps/windows-4.5/teal-001.png'
```

File: tests/testthat/test-shinytest2-filter_panel.R [hunk 3]
```diff
@@ -93,12 +85,31 @@ testthat::test_that("e2e: filtering a module-specific filter is not refected in
...
+  app <- TealAppDriver$new(data = data, modules = example_module())
+
+  # Visible by default
+  filterpanel_accordion_selector <- paste(
+    ".teal-filter-panel [id$=\\'filters-main_filter_accordian\\']",
+    "> .accordion-item > .accordion-collapse"
+  )
+  testthat::expect_true(app$is_visible(filterpanel_accordion_selector))
+  app$click(selector = paste(
+    ".teal-filter-panel [id$='filters-main_filter_accordian']",
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-filter_panel.R]: I fixed the typo `accordian` -> `accordion` in teal.slice regression tests but if you use an id there is no need to specify the class of the element as ids are unique. Start the selector with `#filters-main_filter_accordion` (if https://github.com/insightsengineering/teal.slice/pull/653 gets merged before this PR)

File: tests/testthat/test-shinytest2-module_bookmark_manager.R [hunk 4]
```diff
@@ -56,6 +49,7 @@ testthat::test_that("bookmark_manager_button shows modal with url containing sta
 
   testthat::expect_match(
     rvest::html_text(app$get_html_rvest("div[id$=bookmark_modal] pre")),
-    "_state_id_"
+    "_state_id_=[a-zA-Z0-9]+",
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-module_bookmark_manager.R]: This seems a bit weird. Is this the pattern for ids? Shouldn't then the regex include `-` and `_`? If it is for ids of elements it might accept other characters like `.` or space ` `

File: tests/testthat/test-shinytest2-reporter.R [hunk 5]
```diff
@@ -64,30 +52,20 @@ testthat::test_that("e2e: adding a report card in a module adds it in the report
...
     "Card comment"
   )
-
   app$click(NS(app$active_module_ns(), "reporter-add_report_card_simple-add_card_ok"))
 
+  # Check whether card was added
   app$navigate_teal_tab("Report previewer")
-
   accordian_selector <- sprintf("#%s-pcards .accordion-toggle", app$active_module_ns())
   app$click(selector = accordian_selector)
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-reporter.R]: Her locally it FAILS I'm using the main branch of the dependencies for testing this:
```
Error (test-shinytest2-reporter.R:60:3): e2e: adding a report card in a module adds it in the report previewer tab
Error in `app_find_node_id(self, private, input = input, output = output, 
    selector = selector)`: Cannot find HTML element with selector #teal-teal_modules-nav-temporary_label-module-pcards .accordion-toggle
```

File: tests/testthat/test-shinytest2-wunder_bar.R [hunk 6]
```diff
@@ -35,7 +33,30 @@ testthat::test_that("wunder_bar_srv clicking snapshot icon opens snapshot-manage
...
+
+testthat::test_that("e2e: collapsing wunderbar hides filter and data summary", {
+  skip_if_too_deep(5)
+
+  data <- teal.data::teal_data(mtcars1 = mtcars, mtcars2 = data.frame(am = c(0, 1), test = c("a", "b")))
+  app <- TealAppDriver$new(data = data, modules = example_module())
+  app$click(selector = ".teal_module.active > .bslib-sidebar-layout > button.collapse-toggle")
+  testthat::expect_false(app$is_visible(".teal-filter-panel"))
+  testthat::expect_false(app$is_visible(".teal-active-data-summary-panel"))
+  app$expect_screenshot(selector = ".bslib-sidebar-layout .main")
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-wunder_bar.R:59]: Missing windows snapshot too:

```
Warning (test-shinytest2-wunder_bar.R:49:3): e2e: collapsing wunderbar hides filter and data summary
Adding new file snapshot: 'tests/testthat/_snaps/windows-4.5/teal-001.png'
```

File: tests/testthat/test-shinytest2-show-rcode.R [hunk 7]
```diff
@@ -1,61 +1,78 @@
...
+      app$get_text("#shiny-modal div.modal-header > h4"),
+      "Example Code"
+    )
+
+    # There are two Dismiss buttons with similar id and the same label.
+    dismiss_text <- app$get_text("#shiny-modal button[data-dismiss]")
+    testthat::expect_length(dismiss_text, 2)
+    testthat::expect_setequal(dismiss_text, "Dismiss")
+
+    # Check for Copy buttons.
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-show-rcode.R:40]: This is side comment: but on my computer the `Copy to Clipboard` button doesn't work: The clipboard doesn't collect te R code. Probably it shouldn't be tested but just letting you know in case you know an easy way to test it really works (and then I should check with local IT why it fails on my computer).

File: tests/testthat/test-shinytest2-teal_data_module.R [hunk 8]
```diff
@@ -2,148 +2,72 @@ testthat::skip_if_not_installed("shinytest2")
...
   app$click("teal-data-teal_data_module-submit")
   app$click(selector = "#teal-close_teal_data_module_modal button")
-  testthat::expect_true(
-    app$get_js("document.querySelector('#teal-close_teal_data_module_modal button.disabled') === null")
+
+  testthat::expect_null(
+    app$get_html("#teal-close_teal_data_module_modal button.disabled")
   )
-  testthat::expect_true(app$get_js("document.querySelector('.teal-data-module-popup') === null"))
+  testthat::expect_null(app$get_html(".teal-data-module-popup"))
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-teal_data_module.R]: This is already tested, no need to duplicate tests if the app is the same (Might have missed some other duplicate tests, please check if there are any).

File: tests/testthat/test-shinytest2-teal_data_module.R [hunk 9]
```diff
@@ -282,37 +189,23 @@ testthat::test_that("e2e: teal_data_module gets removed after successful data lo
...
-          data <- within(
-            teal_data(),
-            {
-              dataset1 <- iris
-              dataset2 <- mtcars
-            }
-          )
-
-          data
+          data <- within(teal_data(), {
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-teal_data_module.R]:
```suggestion
          within(teal_data(), {
```

File: tests/testthat/test-shinytest2-teal_data_module.R [hunk 10]
```diff
@@ -325,74 +218,8 @@ testthat::test_that("e2e: teal_data_module is still visible after successful dat
...
-          rvest::html_nodes(
-            app$get_html_rvest("#teal-teal_modules-active_tab"),
-            "a[data-value*='example_module']"
-          ),
-          "disabled"
-        )
-      )
-    )
-  )
+  app$click(selector = "#teal-close_teal_data_module_modal button[data-dismiss='modal']")
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-teal_data_module.R]: Not tested but if you use the id why select with attributes properties too?
```suggestion
  app$click(selector = "#teal-close_teal_data_module_modal")
```

File: tests/testthat/test-shinytest2-wunder_bar.R [hunk 11]
```diff
@@ -35,7 +33,30 @@ testthat::test_that("wunder_bar_srv clicking snapshot icon opens snapshot-manage
...
+})
+
+testthat::test_that("e2e: collapsing wunderbar hides filter and data summary", {
+  skip_if_too_deep(5)
+
+  data <- teal.data::teal_data(mtcars1 = mtcars, mtcars2 = data.frame(am = c(0, 1), test = c("a", "b")))
+  app <- TealAppDriver$new(data = data, modules = example_module())
+  app$click(selector = ".teal_module.active > .bslib-sidebar-layout > button.collapse-toggle")
+  testthat::expect_false(app$is_visible(".teal-filter-panel"))
+  testthat::expect_false(app$is_visible(".teal-active-data-summary-panel"))
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-wunder_bar.R:58]: I think it is best to test also that previously the panel(s) were visible. The teal filter panel could disappear and we wouldn't notice.

File: R/reporter_previewer_module.R [hunk 12]
```diff
@@ -43,6 +43,7 @@ reporter_previewer_module <- function(label = "Report previewer", server_args =
   # This is to prevent another module being labeled "Report previewer".
   class(module) <- c(class(module), "teal_module_previewer")
   module$label <- label
+  module$path <- label
```
Comment (llrs-roche) [inline R/reporter_previewer_module.R:57]: Not sure what this change does, the commit message does not provide any clue and I find surprising that the same information is on two parts of the same object. Should label be in just one place?

File: tests/testthat/test-module_teal.R [hunk 13]
```diff
@@ -466,7 +466,7 @@ testthat::describe("srv_teal teal_modules", {
           `teal_modules-active_module_id` = "module_1"
         )
         out <- modules_output$module_1()
-        testthat::expect_true(!is.null(out))
+        testthat::expect_type(out, "double")
```
Comment (llrs-roche) [inline tests/testthat/test-module_teal.R:469]: What is this magic number and how it is related to "srv_teal teal_modules are not called again when data changes"? All the previous tests expect it to be NULL?

File: tests/testthat/helper-shinytest2.R [hunk 14]
```diff
@@ -35,3 +35,26 @@ report_module <- function(label = "example teal module") {
     }
   )
 }
+
+example_teal_data_module <- function(with_submit = FALSE, once = TRUE) {
```
Comment (llrs-roche) [inline tests/testthat/helper-shinytest2.R]: I only see it being used on `teal_data_module` did you want to reuse it in other tests too?

**Comment by llrs-roche** [comment]
I checked the tests added and I only found 2 bslib regression tests added about collapsing the data summary and the filters panel (no one about collapsing filters of an individual dataset). Not sure if we need to test for other things that might be tested on teal.slice but are modified by teal or some that are teal exclusive regarding bslib.

File: tests/testthat/test-shinytest2-init.R (hunk 2 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-init.R:11]: I uploaded the snap for `linux` using the insightsengineering's rstudio container, I don't see any other window snaps, so I think we should skip this one

File: tests/testthat/test-shinytest2-data_summary.R (hunk 1 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-data_summary.R:99]: This is a `rlang::hash` issue that I also get locally, but not on the tests.

I'm not sure about the best way to address this as it may be a conditional warning, but I'm trying to find a proper solution

File: tests/testthat/test-shinytest2-filter_panel.R (hunk 3 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-filter_panel.R]: Awesome, thanks!

I think this is one of the tests that should be redundant, as it looks for the main filter accordion.

File: tests/testthat/test-shinytest2-module_bookmark_manager.R (hunk 4 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-module_bookmark_manager.R]: I don't think it can start with those as it's based on an hex id:

https://github.com/rstudio/shiny/blob/08383ad8b978ed4add7dedcb3a16929c2610d2d8/R/shiny.R#L26

Comment (llrs-roche) [reply tests/testthat/test-shinytest2-module_bookmark_manager.R]: Then a comment on what this `_state_id_` is would be appreciated by me and possibly future readers of the code.

File: tests/testthat/test-shinytest2-data_summary.R (hunk 1 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-data_summary.R:99]: I invested a bunch of time on this _(not the first time it popups)_ and best solution seems to suppress warning at the `rlang::hash` call

This is not the first time it popups, with https://github.com/insightsengineering/teal.data/issues/351 and https://github.com/insightsengineering/teal.code/issues/194

As well as other times.

The best I can assertain is that `rlang::hash` tries to serialize the namespace included in the `{MAE}` dataset in C++ and it outputs this warning as it is not a base package.

File: tests/testthat/test-module_teal.R (hunk 13 above)
Comment (averissimo) [reply tests/testthat/test-module_teal.R:469]: This is the output of the module (`runif(1)`) defined above.

I changed it from `not null` to an explicit type for the test to be more specific and less prone to keep passing if it starts to return something else (reactive, silent error, etc...)

File: R/reporter_previewer_module.R (hunk 12 above)
Comment (averissimo) [reply R/reporter_previewer_module.R:57]: The reporter is created in a particular way, with a temporary label and then it is changed manually.

This affects the label, but not the path of it. The path appears as a data attribute and this change ensures that it is the same as the label and not `temporary label`

Comment (averissimo) [reply R/reporter_previewer_module.R:57]: It was introduced by the UX/UI redesing, maybe @vedhav can give more context on why it is needed?

My understanding is that is 2-fold, ensuring a unique name while keeping label intact and practically on navigation

File: tests/testthat/helper-shinytest2.R (hunk 14 above)
Comment (averissimo) [reply tests/testthat/helper-shinytest2.R]: It was used several times there and reduced code. I was refraiming from creating a single `helper` file for this function, but there is no real drawback.

`testthat` encourages to keep this kind of functions/constants/etc.. on a helper file.

WDYT about creating `helper-shinytest2-teal_data_module`?

File: tests/testthat/test-shinytest2-show-rcode.R (hunk 7 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-show-rcode.R:40]: Are you saying it doesn't work on any browser or on the rstudio viewer?

File: tests/testthat/test-shinytest2-teal_data_module.R (hunk 10 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-teal_data_module.R]: I believe it's because the button itself needs to be clicked, but I'll confirm

File: tests/testthat/test-shinytest2-show-rcode.R (hunk 7 above)
Comment (llrs-roche) [reply tests/testthat/test-shinytest2-show-rcode.R:40]: I can't install Firefox to test it but on Rstudio viewer I couldn't, and on Chrome I sometimes can't and sometimes I can (just tested on teal.gallery with _stable and _dev and I could in both). But marking as resolved as it might be a RStudio issue

File: tests/testthat/test-shinytest2-filter_panel.R (hunk 3 above)
Comment (llrs-roche) [reply tests/testthat/test-shinytest2-filter_panel.R]: Mmh, so should it be removed from here?

File: tests/testthat/test-shinytest2-data_summary.R (hunk 1 above)
Comment (llrs-roche) [reply tests/testthat/test-shinytest2-data_summary.R:99]: I'll close this comment (and maybe document your finding on one of these issues).

File: R/reporter_previewer_module.R (hunk 12 above)
Comment (llrs-roche) [reply R/reporter_previewer_module.R:57]: Oh yes, now I recall some comments. Nothing to do then (except a comment to document this?)

File: tests/testthat/helper-shinytest2.R (hunk 14 above)
Comment (llrs-roche) [reply tests/testthat/helper-shinytest2.R]: As Dawid commented recently, it would make the tester to chase to multiples files to get a picture of what is being tested. I prefer to have it here or on the file it is used (which is why I pointed out that it wasn't reused elsewhere). I'll leave it up to you but resolving this.

File: tests/testthat/test-shinytest2-filter_panel.R (hunk 3 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-filter_panel.R]: I meant that it should be also here. Regardless, of the tests on `teal.slice`.

As here it looks for its presence and whether it can be collapsed

File: tests/testthat/test-shinytest2-show-rcode.R (hunk 7 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-show-rcode.R:40]: `rstudio` doesn't allow for this to work, regardless of the operating system.

Good to know that you tested on the `teal.gallery`, those are the "truth" :-)

File: tests/testthat/helper-shinytest2.R (hunk 14 above)
Comment (averissimo) [reply tests/testthat/helper-shinytest2.R]: Ok, unless I can find a use case opportunity, I'll move it back to the file.

Keeping the comment unresolved until it's commited

File: R/reporter_previewer_module.R (hunk 12 above)
Comment (averissimo) [reply R/reporter_previewer_module.R:57]: Comment then! For both lines :sweat_smile:  (I updated the comment above that chunk) cfbfd844d

File: tests/testthat/test-shinytest2-teal_data_module.R (hunk 10 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-teal_data_module.R]: We need the `button` in the selector, but good catch on not needing the `data-dismiss`. We should only have 1 button there.

```suggestion
  app$click(selector = "#teal-close_teal_data_module_modal button")
```

File: tests/testthat/test-shinytest2-reporter.R (hunk 5 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-reporter.R]: Can you try again? The `accordian` -> `accordion` typo was corrected here.

Maybe you had the `teal.slice` branch loaded? I'm hoping it's that, because I've run this multiple times in a row and found no problems :see_no_evil:

Comment (llrs-roche) [reply tests/testthat/test-shinytest2-reporter.R]: I tested again and it failed with the same error message. But I think is because I am using `main`.

File: tests/testthat/test-shinytest2-wunder_bar.R (hunk 11 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-wunder_bar.R:58]: I agree, but it should live on its own test.

Added 1 test above with d86b5141a

File: tests/testthat/test-shinytest2-teal_data_module.R (hunk 8 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-teal_data_module.R]: It is, great eye!! I further improved so that the same selector is used instead of duplicating the string

File: tests/testthat/test-shinytest2-filter_panel.R [hunk 15]
```diff
@@ -93,12 +85,31 @@ testthat::test_that("e2e: filtering a module-specific filter is not refected in
...
+testthat::test_that("e2e: filter panel UI can be collpased and expanded (`bslib` regression)", {
+  skip_if_too_deep(5)
+
+  data <- teal.data::teal_data(mtcars1 = mtcars, mtcars2 = data.frame(am = c(0, 1), test = c("a", "b")))
+  app <- TealAppDriver$new(data = data, modules = example_module())
+
+  # Visible by default
+  filterpanel_accordion_selector <- paste(
+    ".teal-filter-panel [id$=\\'filters-main_filter_accordion\\']",
+    "> .accordion-item > .accordion-collapse"
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-filter_panel.R]: Selector could be simplified:
```suggestion
    "#filters-main_filter_accordion > .accordion-item > .accordion-collapse"
```

File: tests/testthat/test-shinytest2-filter_panel.R [hunk 16]
```diff
@@ -93,12 +85,31 @@ testthat::test_that("e2e: filtering a module-specific filter is not refected in
...
+  )
+  testthat::expect_true(app$is_visible(filterpanel_accordion_selector))
+  app$click(selector = paste(
+    ".teal-filter-panel [id$='filters-main_filter_accordion']",
+    "> .accordion-item > .accordion-header button"
```
Comment (llrs-roche) [inline tests/testthat/test-shinytest2-filter_panel.R]:
```suggestion
    "#filters-main_filter_accordion > .accordion-item > .accordion-header button"
```

File: DESCRIPTION [hunk 17]
```diff
@@ -81,7 +81,7 @@ RdMacros:
     lifecycle
 Remotes:
     insightsengineering/teal.reporter@main,
-    insightsengineering/teal.slice@main,
+    insightsengineering/teal.slice@651_regr_test@main,
```
Comment (averissimo) [inline DESCRIPTION]: Works as a blocking comment for PR merge until https://github.com/insightsengineering/teal.slice/pull/653 is merged

```suggestion
    insightsengineering/teal.slice@main,
```

File: tests/testthat/test-shinytest2-filter_panel.R (hunk 15 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-filter_panel.R]: That `id` is not valid, the full id of the element is `"#teal-teal_modules-nav-example_teal_module-filter_panel-filters-main_filter_accordion"`

I've deliberately used the `[id$='...']` just to match the termination of the id, as it becomes very hardcoded and the test and sensitive to `teal` HTML namespace.

@llrs-roche Do you think we should use it fully?

To simplify it a bit, we could also remove the `.teal-filter-panel` prefix

The reason for including in the first place is to have a full match in the selector (to complement the partial id match)

Comment (llrs-roche) [reply tests/testthat/test-shinytest2-filter_panel.R]: Sorry, realized after I made the comments that here the selector used `$=` and `\\` to scape the quotes but I did't look into it deeper.  
I think it is good to keep full id as it is easier to reason. All the nested structure (`>... > ...` ) is already very sensitive to teal structure so I don't think that depending on the namespace would be worse. 
In other tests there was a helper to create the id within the namespace 

IMHO all or most of the selector needed should have an id (or at most on the parent div).

File: DESCRIPTION (hunk 17 above)
Comment (llrs-roche) [reply DESCRIPTION]: PR on teal.slice is merged. Revert to main and merge?

File: tests/testthat/test-shinytest2-filter_panel.R (hunk 15 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-filter_panel.R]: Check the latest commit: a8ce7e6fe

```diff
diff --git a/tests/testthat/test-shinytest2-filter_panel.R b/tests/testthat/test-shinytest2-filter_panel.R
index c38d236ad..592e2fd95 100644
--- a/tests/testthat/test-shinytest2-filter_panel.R
+++ b/tests/testthat/test-shinytest2-filter_panel.R
@@ -100,15 +100,10 @@ testthat::test_that("e2e: filter panel UI can be collpased and expanded (`bslib`
   app <- TealAppDriver$new(data = data, modules = example_module())
 
   # Visible by default
-  filterpanel_accordion_selector <- paste(
-    ".teal-filter-panel [id$=\\'filters-main_filter_accordion\\']",
-    "> .accordion-item > .accordion-collapse"
-  )
+  filter_panel_id <- "#teal-teal_modules-nav-example_teal_module-filter_panel-filters-main_filter_accordion"
+  filterpanel_accordion_selector <- paste(filter_panel_id, "> .accordion-item > .accordion-collapse")
   testthat::expect_true(app$is_visible(filterpanel_accordion_selector))
-  app$click(selector = paste(
-    ".teal-filter-panel [id$='filters-main_filter_accordion']",
-    "> .accordion-item > .accordion-header button"
-  ))
+  app$click(selector = paste(filter_panel_id, "> .accordion-item > .accordion-header button"))
   testthat::expect_false(app$is_visible(filterpanel_accordion_selector))
 
   app$stop()
```
